portfolio/src/main/webapp/script.js

```diff
@@ -13,65 +13,72 @@
+   renderComments(messages);
+}
+
+function renderComments(messages){
+    const messagesListElement = document.getElementById('comments-container');
     if(messagesListElement.innerHTML !== ''){
         messagesListElement.innerHTML = '';
     }
 
-    for(i = 0; i < messages.length; ++i){
-        childToAppend = createListElement(messages[i].propertyMap.email, messages[i].propertyMap.content);
-        childToAppend.appendChild(createButton(messages[i].propertyMap.messageId));
-        messagesListElement.appendChild(childToAppend);
+    for(var i = 0; i < messages.length; ++i){
+        var domListElement = createDOMListElement(messages[i].propertyMap.email, messages[i].propertyMap.content);
+        domListElement.appendChild(createDOMButton(messages[i].propertyMap.messageId));
+        messagesListElement.appendChild(domListElement);
     }
     
     if(messages.length === 0){
-        var content = "No messages available!";
-        childToAppend = createListElement(content);
-        messagesListElement.appendChild(childToAppend);
+        var content = "No comments available!";
+        var domListElement = createDOMListElement(content);
+        messagesListElement.appendChild(domListElement);
     }
-  });
 }
 
-function createListElement(email, text) {
-  const liElement = document.createElement('li');
-  if(email != "")
-   liElement.innerText = email + ": ";
-  liElement.innerText += text;
-  liElement.classList.add('list-group-item');
-  return liElement;
+function createDOMListElement(email, text) {
+  const domListElement = document.createElement('li');
+  if(email !== "")
+   domListElement.innerText = email + ": ";
+  domListElement.innerText += text;
+  domListElement.classList.add('list-group-item');
+  return domListElement;
 }
 
-function createButton(messageId) {
-  const buttonElement = document.createElement('button');
-  buttonElement.classList.add('delete-button');
-  buttonElement.textContent = "Delete";
-  buttonElement.id = messageId;
-  buttonElement.setAttribute( "onClick", "javascript: deleteMessage(this);" );
-  return buttonElement;
+function createDOMButton(messageId) {
+  const domButtonElement = document.createElement('button');
+  domButtonElement.classList.add('delete-button');
+  domButtonElement.textContent = "Delete";
+  domButtonElement.id = messageId;
+  domButtonElement.setAttribute( "onClick", "javascript: handleDeleteCommentClick(this);" );
+  return domButtonElement;
 }
 
+function handleDeleteCommentClick(buttonElement){
+   deleteComment(buttonElement);
+   const messages = getCommentsFromServer();
+   renderComments(messages);
+}
 
-function deleteMessage(thisButton) {
+function deleteComment(thisButton) {
   const params = new URLSearchParams();
   var messageId = thisButton.id;
   var liElementToDelete = thisButton.parentNode;
   params.append('messageId', messageId);
-  deleteLi(liElementToDelete);
   fetch('/delete-data', {method: 'POST', body: params});
```

Comment:
Please return the `fetch` Promise, this way, you can wait for the comment to be deleted, before fetching new comments. This will prevent race conditions, for example when the HTTP request to `/delete-data` takes longer than the call to `/data`.

```
...
return fetch('/delete-data', ....);
```

At the `handleDeleteCommentClick` function, you'll now have to wait for this call to have finished.

```
async function handleDeleteCommentClick(buttonElement){
   await deleteComment(buttonElement);
   const messages = await getCommentsFromServer();
   renderComments(messages);
}
```

Reply:
Solved